tests/CMakeLists.txt

```diff
@@ -0,0 +1,12 @@
+add_subdirectory(catch)
+
+add_executable(tests
```

Comment:
only the deprecated tests should be seeing `test_main.cpp` and `test_utilities`, so I think they shouldn't be referenced outside of `tests/deprecated/`